_posts/2020-07-06-whats-so-super-about-super-tdma.md

```diff
@@ -0,0 +1,226 @@
+
+```groovy
+//! Simulation: Equilateral triangle network
+///////////////////////////////////////////////////////////////////////////////
+///
+/// To run simulation:
+///   bin/unet samples/super-tdma/e3-network
+/// OR
+///   click on the Run button (▶) in UnetSim
+///
+/// Output trace file: logs/trace.nam
+///
+/// Reference:
+/// [1] M. Chitre, M. Motani, and S. Shahabudeen, "Throughput of networks
+///     with large propagation delays", IEEE Journal of Oceanic Engineering,
+///     37(4):645-658, 2012.
+///
+///////////////////////////////////////////////////////////////////////////////
+
+import org.arl.fjage.*
+import org.arl.unet.*
+import org.arl.unet.phy.*
+import static org.arl.unet.Services.*
+
+///////////////////////////////////////////////////////////////////////////////
+// settings
+
+def slot = 423.ms               // default packet length is about 345 ms
+def range = 650.m               // about slot x 1540 m/s
+def time = 15.minutes           // simulation time
+def schedule = [[2, 3, 0, 0],   // schedule from [1]
+                [0, 0, 1, 3],
+                [0, 1, 0, 2]]
+
+///////////////////////////////////////////////////////////////////////////////
+// display documentation
+
+println """
+Equilateral triangle network
+----------------------------
+Internode distance:     ${range} m
+Slot length:            ${(1000*slot).round()} ms
+Simulation time:        ${time} s"""
+
+///////////////////////////////////////////////////////////////////////////////
+// simulate schedule
+
+simulate time, {
+
+  def n = []
+  n << node('1', address: 1, location: [0, 0, 0])
+  n << node('2', address: 2, location: [range, 0, 0])
+  n << node('3', address: 3, location: [0.5*range, 0.866*range, 0])
+
+  n.eachWithIndex { n1, i ->
+    n1.startup = {
+      def phy = agentForService PHYSICAL
+      phy[Physical.DATA].frameLength = phy[Physical.CONTROL].frameLength
+      add new TickerBehavior(1000*slot, {
+        def slen = schedule[i].size()
+        def s = schedule[i][(tickCount-1)%slen]
+        if (s) phy << new TxFrameReq(to: s, type: Physical.DATA)
+      })
+    }
+  }
+
+}
+
+// display statistics
+println """TX:                     ${trace.txCount}
+RX:                     ${trace.rxCount}
+Offered load:           ${trace.offeredLoad.round(3)}
+Throughput:             ${trace.throughput.round(3)}"""
+```
+Let us go through some of the code snippets from the above script to try and understand different components of the implementation.
+
+First we set the distance between the nodes to be 650 m. With sound speed in water to be approximately 1540 m/s, the propagation delay on any link turns out to be 423 ms. Therefore, the slot duration is set to 423 ms. The simulation time is set to 15 minutes and the high-throughput schedule structured as a matrix is given as an input to all three nodes in the network. 
```

Comment:
If we do not indicate shallow/well mixed/isovelocity, this may come across as odd, since many readers may be familiar with deep waters where taking sound speed profile is the norm.

Reply:
Agreed. But, such information is generally abstracted out for network simulations as is the case here. To not digress from the topic at hand, my opinion is to skip this explanation. Let me know otherwise.

Reply:
If it is normal to abstract this in simulations (as long as it is safe to assume our audience knows it), let's go with this.